esp-wifi/src/wifi/mod.rs

```diff
@@ -704,19 +707,87 @@ unsafe extern "C" fn coex_register_start_cb(
     0
 }
 
-pub fn wifi_start_scan(block: bool) -> i32 {
-    let scan_time = wifi_scan_time_t {
-        active: wifi_active_scan_time_t { min: 10, max: 20 },
-        passive: 20,
+#[derive(Clone, Copy)]
+pub enum ScanTypeConfig {
+    Active { min: Duration, max: Duration },
+    Passive(Duration),
+}
+
+impl Default for ScanTypeConfig {
+    fn default() -> Self {
+        Self::Active {
+            min: Duration::from_millis(10),
+            max: Duration::from_millis(20),
+        }
+    }
+}
+
+#[derive(Clone, Copy, Default)]
+pub struct ScanConfig<'a> {
+    pub ssid: Option<&'a str>,
+    pub bssid: Option<[u8; 6]>,
+    pub channel: Option<u8>,
+    pub show_hidden: bool,
+    pub scan_type: ScanTypeConfig,
+}
+
+pub fn wifi_start_scan(block: bool, config: ScanConfig<'_>) -> i32 {
+    let ScanConfig {
+        ssid,
+        mut bssid,
+        channel,
+        show_hidden,
+        scan_type,
+    } = config;
+    let (scan_time, scan_type) = match scan_type {
+        ScanTypeConfig::Active { min, max } => (
+            wifi_scan_time_t {
+                active: wifi_active_scan_time_t {
+                    min: min.as_millis() as u32,
+                    max: max.as_millis() as u32,
+                },
+                passive: 0,
+            },
+            wifi_scan_type_t_WIFI_SCAN_TYPE_ACTIVE,
+        ),
+        ScanTypeConfig::Passive(dur) => (
+            wifi_scan_time_t {
+                active: wifi_active_scan_time_t { min: 0, max: 0 },
+                passive: dur.as_millis() as u32,
+            },
+            wifi_scan_type_t_WIFI_SCAN_TYPE_PASSIVE,
+        ),
     };
 
+    if !ssid
+        .map(|e| e.chars().all(|e| e.is_ascii()))
+        .unwrap_or(true)
+    {
+        panic!("SSID muste be ASCII");
```

Comment:
```suggestion
        panic!("SSID must be ASCII");
```

Reply:
This isn't correct, though. The SSID isn't necessarily ASCII, it can be anything. It's usually ASCII, but it's not a requirement.

Reply:
I have removed this